orangecontrib/geo/widgets/owgeotransform.py

```diff
@@ -40,107 +56,106 @@ class Outputs:
-
-        geo_args = dict(contentsLength=100, searchable=True,
-                        model=self.variable_model, orientation=Qt.Horizontal)
-
-        gui.comboBox(box, self, "attr_lat", label="Latitude:", **geo_args)
-        gui.comboBox(box, self, "attr_lon", label="Longitude:", **geo_args)
-
-        box = gui.vBox(self.controlArea, box="Transformation:")
-
-        transform_model = PyListModelTooltip()
-        transform_model.setParent(self)
-        transform_model[:] = chain(list(self.EPSG_CODES))
-        transform_model.tooltips[:] = chain(list(self.EPSG_CODES))
-
-        proj_args = dict(contentsLength=100, searchable=True,
-                         model=transform_model, orientation=Qt.Horizontal,
-                         sendSelectedValue=True)
-
-        gui.comboBox(box, self, "from_idx", label="From:", **proj_args)
-        gui.comboBox(box, self, "to_idx", label="To:", **proj_args)
+        args = dict(contentsLength=100, searchable=True,
+                    model=self.variable_model, orientation=Qt.Horizontal)
+        layout.addRow("Latitude:", gui.comboBox(None, self, "attr_lat", **args))
+        layout.addRow("Longitude:", gui.comboBox(None, self, "attr_lon", **args))
+        layout.addWidget(
+            gui.checkBox(
+                None, self, "replace_original", "Replace original coordinates",
+                tooltip="If unchecked, the original coordinates are retained "
+                        "and new coordinates are added as separate variables."))
+
+        layout = QFormLayout()
+        gui.widgetBox(self.controlArea, "Transformation:", orientation=layout)
+        args["model"] = PyListModel(self.EPSG_CODES)
+        layout.addRow("From:", gui.comboBox(None, self, "from_idx", **args))
+        layout.addRow("To:", gui.comboBox(None, self, "to_idx", **args))
 
         self.commit_button = gui.button(self.controlArea, self, "&Commit",
                                         self.apply)
 
     def init_attr_values(self):
-        self.Error.clear()
-        lat, lon = None, None
-        if self.data is not None:
+        self.variable_model.set_domain(self.data.domain if self.data else None)
+        self.Error.no_lat_lon_vars.clear()
+
+        if self.variable_model.rowCount() < 2:
+            self.Error.no_lat_lon_vars()
+            lat, lon = None, None
+            self.data = None
+        else:
             lat, lon = find_lat_lon(self.data, filter_hidden=True)
             if lat is None or lon is None:
-                # we either find both or we try all numeric
-                if self.data.domain.has_continuous_attributes(
-                        include_metas=True):
-                    vars = [var for var in self.data.domain.variables
-                            if var.is_continuous] + \
-                           [var for var in self.data.domain.metas if
-                            var.is_continuous]
-                    lat, lon = vars[0], vars[0]
```

Comment:
This was probably unintentional - we probably want to use the first two, not the first one twice.

Reply:
Actually, it was intentional in case there's only one numeric variable. The user will have to set proper variables anyway in this case.

Reply:
Make sense. What about this?

- If there are less than two, complain, report an error and show nothing.
- If there are two, select them; in at least 50 % cases the user won't need to do anything, in the remaining (s)he will have to switch.
- If there are more than two, don't guess but select `vars[0], vars[0]`.

Reply:
I've got a couple of adjustments I'd like to make to GeoTransform anyway, so I suggest we merge this as is and include improvements in a subsequent PR.